File: 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int idx = 0;
        int prof = 0;
        for(int i = 1; i< prices.size(); i++)
        {
            if(prices[i] - prices[idx] < 0)
            {
                idx = i;
            }
            else
            {
                if(prices[i] - prices[idx] > prof)
                    prof = prices[i] - prices[idx];
            }
        }
        return prof;
    }
};

File: 121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minimum, profit = float('inf'), 0
        for price in prices:
            if price < minimum:
                minimum = price
            profit = max(price - minimum, profit)
        return profit
